### ISSUE TITLE
ImportError: libcurl.so.4: cannot open shared object file: No such file or directory

### ISSUE THREAD

**Issue by gianfilippo** [bioinform/somaticseq#72]
Hi again,

I am having another issue with a libcurl.so.4 import error.
Start at 2019/10/27 15:20:45
Traceback (most recent call last):
  File "/usr/local/bamsurgeon/scripts/sortedBamSplit.py", line 4, in <module>
    import pysam, random, argparse, sys
  File "/softwarepath/Python/2.7.13-foss-2016b/lib/python2.7/site-packages/pysam-0.10.0-py2.7-linux-x86_64.egg/pysam/__init__.py", line 5, in <module>
    from pysam.libchtslib import *
ImportError: libcurl.so.4: cannot open shared object file: No such file or directory

This happens at the following command in the (bamsurgeon) file your script generates
singularity exec --bind /:/mnt   docker://lethalfang/bamsurgeon:1.1-3 \
/usr/local/bamsurgeon/scripts/sortedBamSplit.py \
--bam /mnt//home/Test/Sample_G1700T_012/1/TNMerged.bam \
--proportion 0.5 \
--downsample 1 \
--pick1 /mnt//home/Test/Sample_G1700T_012/1/Designated.Normal.bam \
--pick2 /mnt//home/Test/Sample_G1700T_012/1/Designated.Tumor.bam \
--seed 0

It seems it is trying to run pycurl from a Python 2.7, while everything else runs from python 3.x

Do you think this is some env variable that I have to set or some sort of inconsistency in the docker file ?
Unfortunately I am not very familiar with either docker or singularity

Thanks

**Comment by ltfang-bina** [comment]
Singularity does import some of your local environmental variables.... though I think BamSurgeon actually runs from python2. I'll take a look what's up next week to see what's up. I don't use singularities often so I can't make promises.

**Comment by gianfilippo** [comment]
Hi,

did you have a chance to look into it ?

I am seeing some other problem. I will open a new issue for that.

Thanks for your help!

**Comment by ltfang-bina** [comment]
Would `--cleanenv` flag fix the issue, i.e., `singularity exec --cleanenv....`?

**Comment by gianfilippo** [comment]
Hi,

I can try, but where would I use the command ? add it to a script or just executing it before starting the pipeline ?

Thanks
Gianfilippo

**Comment by ltfang-bina** [comment]
Try it for a command first, e.g., 
```singularity exec --cleanenv --bind /:/mnt docker://lethalfang/bamsurgeon:1.1-3
/usr/local/bamsurgeon/scripts/sortedBamSplit.py
--bam /mnt//home/Test/Sample_G1700T_012/1/TNMerged.bam
--proportion 0.5
--downsample 1
--pick1 /mnt//home/Test/Sample_G1700T_012/1/Designated.Normal.bam
--pick2 /mnt//home/Test/Sample_G1700T_012/1/Designated.Tumor.bam
--seed 0
```

If it works, I can add it to all singularity commands.

**Comment by gianfilippo** [comment]
Hi,
I added the --cleanenv to all the singularity exec commands.
The BamSimulator script generated by somaticseq runs and generates the 
Designated.Tumor.bam
Designated.Normal.bam

After that, the random_sINDEL.bed  random_sSNV.bed  random_sSV.bed are all empty and eventually the pipeline stops with the following error
Error: The requested file (/mnt//home/project/Train_hg38_BWA/Sample_G1700T_012/4/cnvfile.bed) could not be opened. Error message: (No such file or directory). Exiting!

Before the error message I get a number of warnings:
WARNING: Non existent 'bind path' source: '/ycga-ba'
WARNING: Non existent 'bind path' source: '/ycga-ba'
WARNING: Non existent 'bind path' source: '/ycga-ba'
WARNING: Non existent 'bind path' source: '/ycga-ba'
Traceback (most recent call last):
  File "/usr/local/bamsurgeon/scripts/randomsites.py", line 210, in <module>
    main(args)
  File "/usr/local/bamsurgeon/scripts/randomsites.py", line 173, in main
    g = Genome(args.genome, bedfile=args.bed)
  File "/usr/local/bamsurgeon/scripts/randomsites.py", line 40, in __init__
    start = int(start)
ValueError: invalid literal for int() with base 10: '6587354.0'
WARNING: Non existent 'bind path' source: '/ycga-ba'
Traceback (most recent call last):
  File "/usr/local/bamsurgeon/scripts/randomsites.py", line 210, in <module>
    main(args)
  File "/usr/local/bamsurgeon/scripts/randomsites.py", line 173, in main
    g = Genome(args.genome, bedfile=args.bed)
  File "/usr/local/bamsurgeon/scripts/randomsites.py", line 40, in __init__
    start = int(start)
ValueError: invalid literal for int() with base 10: '6587354.0'
Use of uninitialized value $header in print at /usr/local/bin/vcfsorter.pl line 82.
WARNING: Non existent 'bind path' source: '/ycga-ba'
Traceback (most recent call last):
  File "/usr/local/bamsurgeon/scripts/randomsites.py", line 210, in <module>
    main(args)
  File "/usr/local/bamsurgeon/scripts/randomsites.py", line 173, in main
    g = Genome(args.genome, bedfile=args.bed)
  File "/usr/local/bamsurgeon/scripts/randomsites.py", line 40, in __init__
    start = int(start)
ValueError: invalid literal for int() with base 10: '6587354.0'
Use of uninitialized value $header in print at /usr/local/bin/vcfsorter.pl line 82.
WARNING: Non existent 'bind path' source: '/ycga-ba'

What do you think ?

Thanks
Gianfilippo

**Comment by gianfilippo** [comment]
Hi,

this issue is now resolved. It turns out it was some lib issue, not related to somaticseq

Thanks
